src/ScheduledJobs/Repositories/ParticipantRegistrationDynamoDbRepository.cs

```diff
@@ -24,14 +25,39 @@ public ParticipantRegistrationDynamoDbRepository(IAmazonDynamoDB client, IDynamo
             _config = new DynamoDBOperationConfig { OverrideTableName = awsSettings.ParticipantRegistrationDynamoDbTableName };
         }
 
-        public async Task<IEnumerable<Participant>> GetAllAsync()
+        public Task<IAsyncEnumerable<Participant>> GetAllAsync()
         {
-            return await _context.ScanAsync<Participant>(null, _config).GetRemainingAsync();
+            return Task.FromResult(GetAllAsyncEnumerable());
         }
 
+        public async IAsyncEnumerable<TModel> GetAllMappedAsync<TModel>(Func<Participant, TModel> mapper)
+        {
+            await foreach (var participant in GetAllAsyncEnumerable())
+            {
+                yield return mapper(participant);
```

Comment:
No need to do the mapping here, just get and `IEnumerable<Participant>` and then do `.Select(x=>mapper(x))` when needed.